runtime/hydradx/src/evm/mod.rs

```diff
@@ -139,14 +141,35 @@ parameter_types! {
 	pub const SuicideQuickClearLimit: u32 = 0;
 }
 
+pub struct EnsureAddressMappedOrTruncated<M>(sp_std::marker::PhantomData<M>);
+
+impl<M, OuterOrigin> EnsureAddressOrigin<OuterOrigin> for EnsureAddressMappedOrTruncated<M>
+where
+	M: AddressMapping<AccountId32>,
+	OuterOrigin: Into<Result<RawOrigin<AccountId32>, OuterOrigin>> + From<RawOrigin<AccountId32>>,
+{
+	type Success = AccountId32;
+
+	fn try_address_origin(address: &sp_core::H160, origin: OuterOrigin) -> Result<AccountId32, OuterOrigin> {
+		origin.into().and_then(|o| match o {
+			RawOrigin::Signed(who)
+				if who == M::into_account_id(*address) || AsRef::<[u8; 32]>::as_ref(&who)[0..20] == address[0..20] =>
```

Comment:
potencially only `truncated_account_id(address)` would be needed here, bound accounts should be already covered

Reply:
also would be nice to have some test for this case